## Урок 1. Случайные события. Условная вероятность. Формула Байеса. Независимые испытания

#### 1. Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется ХОТЯ бы один туз.

#### а.
Т.к. в колоде всего 4 масти, то вероятность того что одна карта окажется крести равна 1/4.
Тогда вероятность того что все 4 карты крести равна произведению их вероятностей:

In [12]:
1/4 * 1/4 * 1/4 * 1/4

0.00390625

Если доставать карты последовательно, то вероятность каждой последующей карты будет уменьшаться, т.к. предполагается, что каждый раз будем извлекать карту крести:

In [13]:
13/52 * 12/51 * 11/50 * 10/49

0.0026410564225690276

#### б.

Т.к. в колоде 4 туза, то вероятность одного туза равна 4/52 = 1/13.
Условие "хотя бы один" допускает что может выпасть 1, 2, 3 или 4 туза, поэтому их вероятность равна сумме вероятностей:

In [14]:
1/13 + 1/13 + 1/13 + 1/13

0.3076923076923077

Если доставать карты последовательно, то для того чтобы выполнить условие "хотя бы один", нужно исходить из предположения, что каждый раз будем извлекать из колоды не туза (вероятность каждой последующей карты будет увеличиваться):

In [18]:
4/52 + 4/51 + 4/50 + 4/49

0.316987102533321

#### 2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

Воспользуемся формулой из комбинаторики для вычисления числа размещений (т.к. в данном случае порядок не важен).

In [3]:
from math import factorial

In [4]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [5]:
combinations(10, 3)

120

Т.е. всего возможно 120 комбинаций. Тогда вероятность открыть дверь с первой попытки равна 1/120

#### 3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

Вероятность извлечь одну окрашенную деталь равна 9/15, или 3/5. 
Если достать детали одновременно, то:

In [19]:
3/5 * 3/5 * 3/5

0.21600000000000003

Если последовательно:

In [21]:
9/15 * 8/14 * 7/13

0.1846153846153846

#### 4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

Вероятность одного выигрышного билета 1/50.
При одновременном приобретении 2-х билетов вероятность получится:

In [23]:
1/50 * 1/50

0.0004

Если последовательно:

In [24]:
2/100 * 1/99

0.00020202020202020202